### ISSUE TITLE
JTextField Highlight Color Issue

### ISSUE THREAD

**Issue by srcarter3** [vincenzopalazzo/material-ui-swing#52]
I have the following code which overrides default settings on a text field: 
`UIManager.put("TextField.font", theme.getRegular());
UIManager.put("TextField.background", theme.getHighlightBackgroundPrimary());
UIManager.put("TextField.foreground", theme.getTextColor());
UIManager.put("TextField.caretForeground", theme.getTextColor());
UIManager.put("TextField.selectionForeground", theme.getTextColor());
UIManager.put("TextField.inactiveForeground", theme.getTextColor());
UIManager.put("TextField.selectionBackground", theme.getTooltipHighlight());`

This produces a good result, the background is a gray, the underline is white when not selected, orange when selected.  The text is white.  The highlight is orange.

On Linux, when I highlight a textfield (highlight is orange, text is white), and then move to a new textfield (either by tabbing, or clicking into a new field) then the previous text now has a light gray highlight with white text. It should just go back to a normal look (gray background, white underline, white text).
Any idea what is causing this?  I'm assuming their is some library setting that is changing the state of the colors.
This problem does not occur on my Windows machine, however.
![Screenshot-Add CRISM stamp layer](https://user-images.githubusercontent.com/21114460/58292437-34629680-7d76-11e9-91a3-d8c1f15b6647.png)
![Screenshot-Add CRISM stamp layer-1](https://user-images.githubusercontent.com/21114460/58292436-34629680-7d76-11e9-9eba-f7feedc410b7.png)
![Screenshot-Add CRISM stamp layer-2](https://user-images.githubusercontent.com/21114460/58292435-34629680-7d76-11e9-91af-fdaa0dfffd51.png)

**Comment by vincenzopalazzo** [comment]
HI @srcarter3,
The text field and the password field was rewriting for fix your bug, I use Linux and not have this problem. 

a correction the  constant UI are:

```
 table.put("TextField.background", MaterialColors.GRAY_100);
 table.put("TextField.foreground", MaterialColors.GRAY_800);
 table.put("TextField.inactiveForeground", MaterialColors.GRAY_800);
 table.put("TextField.inactiveBackground", MaterialColors.GRAY_200);
 table.put("TextField.selectionBackground", MaterialColors.LIGHT_BLUE_400);
 table.put("TextField.selectionForeground", MaterialColors.BLACK);
 table.put("TextField.border", BorderFactory.createEmptyBorder(3, 5, 2, 5));
```

Not exist the `UIManager.put("TextField.caretForeground", theme.getTextColor()); `

But I look this bug in the weekend

**Comment by vincenzopalazzo** [comment]
Hi @srcarter3,

The solution is: `UIManager.put("TextField.inactiveBackground",  theme.getRegular());`

Is strange that your code work ok on Windows, but with this constant, you should resolve your problem

**Comment by zanderson9** [comment]
Hi @vincenzopalazzo,
The solution that you suggested is not correct. Mainly because the value for the key "TextField.inactiveBackground" is "**Color**", whereas our theme.getRegular() API call returns "**Font**", 
So what exactly did you mean for the value of "TextField.inactiveBackground"?

**Comment by vincenzopalazzo** [comment]
Hi @zanderson9,

Sorry for my wrong response,  the `inactiveBackground` is the color of the component takes when is not selection, I tested the look and feel with your configuration and I have the same problem with UIManager.put("TextField.caretForeground", yourColor); resolve the problem.

**Comment by srcarter3** [comment]
Hi @vincenzopalazzo,

Setting the inactiveBackground to be the same as the background causes the underline to disappear. See the screenshot below:

![Screenshot-Add Projected CTX stamp layer](https://user-images.githubusercontent.com/21114460/58504257-0519a400-813f-11e9-8932-62dbce024e36.png)

**Comment by vincenzopalazzo** [comment]
HI @srcarter3,

Can you give me the configuration?

**Comment by srcarter3** [comment]
All I did was add a setting for the inactive background:
`UIManager.put("TextField.inactiveBackground", theme.getHighlightBackgroundPrimary());`

**Comment by vincenzopalazzo** [comment]
@srcarter3 

I try this config into my personal university project, this is the effect

When the text field is active
![active](https://user-images.githubusercontent.com/17150045/58507506-366f8f80-8192-11e9-99e9-262dbc6d8912.png)

When the text field is selected
![selected](https://user-images.githubusercontent.com/17150045/58507530-425b5180-8192-11e9-8359-c2623364c3a5.png)

when is selected inactive
![selectedInactive](https://user-images.githubusercontent.com/17150045/58507551-4d15e680-8192-11e9-8cfd-8638a6b7670d.png)

when is inactive
![inactived](https://user-images.githubusercontent.com/17150045/58507586-57d07b80-8192-11e9-8229-c16418f8a23e.png)

This is the code for my configuration, if it not work can you give me a complete example and expectations of the work?

```
UIManager.put("TextField.background", MaterialColors.COSMO_BLACK);
UIManager.put("TextField.foreground", MaterialColors.WHITE);
UIManager.put("TextField.inactiveForeground", MaterialColors.WHITE);
UIManager.put("TextField.inactiveBackground", MaterialColors.COSMO_LIGHT_ORANGE);
UIManager.put("TextField.selectionBackground", MaterialColors.COSMO_ORANGE);
UIManager.put("TextField.selectionForeground", MaterialColors.WHITE);
```

**Comment by srcarter3** [comment]
Hi Vincent,

I believe that "selected inactive" is not a valid state.  Once that field is inactive, it should no longer be selected.  "Selected inactive" should look the same as "inactive".

**Comment by vincenzopalazzo** [comment]
Hi @srcarter3 I look the code and I do feedback you, thanks so much

**Comment by vincenzopalazzo** [comment]
Hi @atarw,
I have the need to your help for resolving this bug, look the last two message into this issue, I not found a mode for change color background when the state JTextField change form focus-selected to not-focus-selected, @srcarter3 says that the not-forcus-selected is a not valid status, you have any idea for fix this bug? the code is the last version stable for the library.

This is the gif of the bug
![](https://media.giphy.com/media/9Dt9DX3iDoWOuMRoP2/giphy.gif)

The setting into UIManageri is a simile to

```
UIManager.put("TextField.background", MaterialColors.COSMO_BLACK);
UIManager.put("TextField.foreground", MaterialColors.WHITE);
UIManager.put("TextField.inactiveForeground", MaterialColors.WHITE);
UIManager.put("TextField.selectionBackground", MaterialColors.COSMO_ORANGE);
UIManager.put("TextField.selectionForeground", MaterialColors.WHITE);
```

The constant `TextField.inactiveBackground` is set by default look and feel

**Comment by atarw** [comment]
Due to the terms of my internship I'm unfortunately not allowed to work on outside projects for the summer :/ Sorry about that, I'll be back by August

**Comment by vincenzopalazzo** [comment]
@atarw Don't worry, good work and sorry for the retard in the answare

**Comment by vincenzopalazzo** [comment]
I Fixed it, is available into the developing branch

**Comment by vincenzopalazzo** [comment]
Released version beta 1.3, it fixed this bug
